common/src/main/java/com/wynntils/screens/maps/GuildMapScreen.java

```diff
@@ -252,17 +325,134 @@ public void doRender(GuiGraphics guiGraphics, int mouseX, int mouseY, float part
-    @Override
-    protected Stream<MapFeature> getRenderedMapFeatures() {
-        // FIXME: Add back hybrid/advancement mode
-        // FIXME: Add territory connection map paths
-        // FIXME: Add user markers
-        return Services.MapData.getFeaturesForCategory("wynntils:territory")
+    private void renderTerritoryAreaPaths(PoseStack poseStack) {
+        List<TerritoryArea> territoryAreas = getRenderedMapFeatures()
+                .filter(f -> f instanceof TerritoryArea)
+                .map(f -> (TerritoryArea) f)
+                .toList();
+
+        for (MapFeature feature : territoryAreas) {
+            if (feature instanceof TerritoryArea territoryArea) {
+                TerritoryInfo territoryInfo = Models.Territory.getTerritoryInfo(
+                        territoryArea.getTerritoryProfile().getName());
+                if (territoryInfo == null) continue;
+
+                for (String tradingRoute : territoryInfo.getTradingRoutes()) {
+                    TerritoryArea destination = territoryAreas.stream()
+                            .filter(area -> area.getTerritoryProfile().getName().equals(tradingRoute))
+                            .findFirst()
+                            .orElse(null);
+
+                    if (destination == null) continue;
+
+                    Vector2f firstCentroid = territoryArea.getBoundingPolygon().centroid();
+                    Vector2f secondCentroid = destination.getBoundingPolygon().centroid();
+                    float firstWorldX =
+                            MapRenderer.getRenderX((int) firstCentroid.x(), mapCenterX, centerX, zoomRenderScale);
+                    float firstWorldZ =
+                            MapRenderer.getRenderZ((int) firstCentroid.y(), mapCenterZ, centerZ, zoomRenderScale);
+                    float secondWorldX =
+                            MapRenderer.getRenderX((int) secondCentroid.x(), mapCenterX, centerX, zoomRenderScale);
+                    float secondWorldZ =
+                            MapRenderer.getRenderZ((int) secondCentroid.y(), mapCenterZ, centerZ, zoomRenderScale);
+                    RenderUtils.drawLine(
+                            poseStack, CommonColors.BLACK, firstWorldX, firstWorldZ, secondWorldX, secondWorldZ, 0, 1);
+                }
+            }
+        }
+    }
+
+    private void renderHeadquarterIcons(PoseStack poseStack) {
+        List<TerritoryArea> hqTerritoryAreas = getRenderedMapFeatures()
                 .filter(f -> f instanceof TerritoryArea)
                 .map(f -> (TerritoryArea) f)
-                .filter(this::filterDefense)
-                .filter(this::filterTreasury)
-                .map(f -> f);
+                .filter(territoryArea -> territoryArea
+                        .getTerritoryInfo()
+                        .map(TerritoryInfo::isHeadquarters)
+                        .orElse(false))
+                .toList();
+
+        for (TerritoryArea hqTerritoryArea : hqTerritoryAreas) {
```

Comment:
I have kind of the same idea here (and for timers), it seems like it would be even more MapData-ish to have these as Locations. But, once again, let's leave any such change to a future additional round of cleanup after integration into main.